### PR TITLE
Add toJSON method to Remote Config Template implementation

### PR DESCRIPTION
**Pull request by lahirumaramba**
- Add `toJSON()` method to `RemoteConfigTemplateImpl` to correctly serialize the JSON object

### CODE REVIEW

**Comment by lahirumaramba** [comment]
Adding unit tests, please hold the review.

File: src/remote-config/remote-config.ts [hunk 1]
```diff
@@ -188,4 +188,16 @@ class RemoteConfigTemplateImpl implements RemoteConfigTemplate {
...
+
+  /**
+   * @return {RemoteConfigTemplate} A JSON-serializable representation of this object.
+   */
+  public toJSON(): RemoteConfigTemplate {
+    return {
+      conditions: this.conditions,
+      parameters: this.parameters,
+      parameterGroups: this.parameterGroups,
+      etag: this.etagInternal,
```
Comment (hiranya911) [inline src/remote-config/remote-config.ts:200]: Can we use `this.etag` here?

File: test/unit/remote-config/remote-config.spec.ts [hunk 2]
```diff
@@ -314,6 +314,8 @@ describe('RemoteConfig', () => {
           expect(cond.name).to.equal('ios');
           expect(cond.expression).to.equal('device.os == \'ios\'');
           expect(cond.tagColor).to.equal(TagColor.BLUE);
+
+          expect(JSON.stringify(template)).equals(JSON.stringify(REMOTE_CONFIG_RESPONSE));
```
Comment (hiranya911) [inline test/unit/remote-config/remote-config.spec.ts:318]:
```
const parsed = JSON.parse(JSON.stringify(template));
expect(parsed).deep.equal(REMOTE_CONFIG_RESPONSE);
```

File: src/remote-config/remote-config.ts (hunk 1 above)
Comment (lahirumaramba) [reply src/remote-config/remote-config.ts]: Good catch! Thanks!

File: test/unit/remote-config/remote-config.spec.ts (hunk 2 above)
Comment (lahirumaramba) [reply test/unit/remote-config/remote-config.spec.ts]: Updated!

**Pull request merged**